lib/clientv2/node_e2e_test.go

```diff
@@ -0,0 +1,392 @@
+// Copyright (c) 2017 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clientv2_test
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"context"
+
+	"github.com/projectcalico/libcalico-go/lib/apiconfig"
+	"github.com/projectcalico/libcalico-go/lib/apiv2"
+	"github.com/projectcalico/libcalico-go/lib/backend"
+	"github.com/projectcalico/libcalico-go/lib/clientv2"
+	"github.com/projectcalico/libcalico-go/lib/options"
+	"github.com/projectcalico/libcalico-go/lib/testutils"
+	"github.com/projectcalico/libcalico-go/lib/watch"
+)
+
+var _ = testutils.E2eDatastoreDescribe("Profile tests", testutils.DatastoreAll, func(config apiconfig.CalicoAPIConfig) {
+
+	name1 := "profile-1"
+	name2 := "profile-2"
+	spec1 := apiv2.ProfileSpec{
+		LabelsToApply: map[string]string {
+			"aa": "bb",
+		},
+	}
+	spec2 := apiv2.ProfileSpec{
+		LabelsToApply: map[string]string {
+			"bb": "cc",
+		},
+	}
+
+	DescribeTable("Profile e2e CRUD tests",
+		func(name1, name2 string, spec1, spec2 apiv2.ProfileSpec) {
+			c, err := clientv2.New(config)
+			Expect(err).NotTo(HaveOccurred())
+
+			be, err := backend.NewClient(config)
+			Expect(err).NotTo(HaveOccurred())
+			be.Clean()
+
+			By("Updating the Profile before it is created")
+			res, outError := c.Profiles().Update(context.Background(), &apiv2.Profile{
```

Comment:
these are all the profile operations, the file name is `node_e2e_test.go`

Reply:
arrrgh!

Reply:
I think my two files are the wrong way round lol